src/main/java/com/hpe/adm/octane/ideplugins/services/nonentity/ImageService.java

```diff
@@ -77,22 +84,20 @@ private File saveImageToTempFile(String pictureLink) throws Exception {
      * @return
      */
     private HttpResponse downloadImage(String pictureLink, int tryCount) throws Exception {
-        if (lwssoCookie == null) {
-            lwssoCookie = clientLoginCookieProvider.get();
-        }
 
         HttpResponse httpResponse;
         HttpTransport HTTP_TRANSPORT = new NetHttpTransport();
         HttpRequestFactory requestFactory = HTTP_TRANSPORT.createRequestFactory();
         try {
             HttpRequest httpRequest = requestFactory.buildGetRequest(new GenericUrl(pictureLink));
+            HttpCookie lwssoCookie = ((IdePluginsOctaneHttpClient) httpClientProvider.getOctaneHttpClient()).getSessionHttpCookie();
             httpRequest.getHeaders().setCookie(lwssoCookie.toString());
             httpResponse = httpRequest.execute();
         } catch (IOException e) {
             if(e instanceof HttpResponseException && ((HttpResponseException)e).getStatusCode() == 401 && tryCount > 0) {
                 //means that the cookie expired
                 logger.error("Cookie expired, retrying: " + e.getMessage());
-                lwssoCookie = clientLoginCookieProvider.get();
+                // lwssoCookie = clientLoginCookieProvider.get(); TODO: andras
                 return downloadImage(pictureLink, --tryCount);
```

Comment:
Is something needed here? it is a Todo with no message

Reply:
forgot, i fix